### PR TITLE
DOC: GH27557 Updates Python support documentation to match NumPy NEP 29

### PR DESCRIPTION
**Pull request by JosephParampathu**
- [x] closes [#27557](https://github.com/pandas-dev/pandas/issues/27557)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This updates the documentation for Python support to match NEP 29, and includes a link to NEP 29. The change is at line 410 of docs/development/policies.html.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
umm, just add a link here: https://github.com/pandas-dev/pandas/tree/main/doc/source/development

**Comment by JosephParampathu** [comment]
Thanks, I included the change to the policies.rst document as well. I resubmitted the PR using the double underscore. Is there a way for me to test that a link in an .rst file works in my local environment first? With the .html file, I checked it in my browser, but am I correct in understanding something similar is not possible with .rst files?

File: doc/build/html/development/policies.html [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+   Debugging C extensions
+  </a>
+ </li>
+ <li class="toctree-l1">
+  <a class="reference internal" href="extending.html">
+   Extending pandas
+  </a>
+ </li>
+ <li class="toctree-l1">
+  <a class="reference internal" href="developer.html">
```
Comment (jreback) [inline doc/build/html/development/policies.html:296]: you need to remove this file

**Comment by mroeschke** [comment]
You can test this locally by building the docs with these instructions: https://pandas.pydata.org/pandas-docs/stable/development/contributing_documentation.html#how-to-build-the-pandas-documentation

File: doc/build/html/development/policies.html (hunk 1 above)
Comment (JosephParampathu) [reply doc/build/html/development/policies.html]: I have removed the policies.html file and left the changes only in the policies.rst file. Thank you and mroeschke for the help.

**Review by jreback — APPROVED** [review]

**Pull request merged**